src/lib/OS/PageTable.cc

```diff
@@ -100,17 +100,14 @@ uint64_t PageTable::deleteMapping(uint64_t vaddr, size_t size) {
   std::vector<PageTable::TableItr> itrs;
 
   // Increment down aligned vaddr by PAGE_SIZE every loop iteration until
-  // (size / PAGE_SIZE) number of address ranges have been covered. In each loop
-  // iteration check if address range mapping exists, if not return a page
-  // table fault.
+  // (size / PAGE_SIZE) number of address ranges have been covered.
   while (size > 0) {
     auto itr = table_.find(vaddr);
-    if (itr == table_.end()) {
-      return masks::faults::pagetable::FAULT | masks::faults::pagetable::UNMAP;
```

Comment:
So we shouldn't fault on a failed unmap attempt?

Reply:
@rahahahat please advise on this

Reply:
@jj16791 If you remember we had meeting about this and decided to remove it, it previously did fault but we kept getting the "Mapping does not exist" error during the OMP sprint. This new impl is also in line with the munmap specification.

> The address addr must be a multiple of the page size. All pages containing a part of the indicated range are unmapped, and subsequent references to these pages will generate SIGSEGV. It is not an error if the indicated range does not contain any mapped pages.

Reply:
yes I remember now, thank you.